### PR TITLE
HW Spring 3 added

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImplImpl.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.dev.cinema.dao.impl;
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImplImpl.java:1]: i think you can rename this class from `CinemaHallDaoImplImpl` to `CinemaHallDaoImpl`

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImplImpl.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.dev.cinema.dao.CinemaHallDao;
+import com.dev.cinema.models.CinemaHall;
+import java.util.List;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class CinemaHallDaoImplImpl extends GenericDaoImpl<CinemaHall> implements CinemaHallDao {
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImplImpl.java]:
```suggestion
public class CinemaHallDaoImpl extends GenericDaoImpl<CinemaHall> implements CinemaHallDao {
```
The same for other Dao classes

**Review by AnaLanda — COMMENTED** [review]
Looks good! 😃

File: pom.xml [hunk 3]
```diff
@@ -44,6 +65,26 @@
...
+            <version>2.8.0</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
+            <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (RomanDubovskyi) [inline pom.xml]: do we have any jsp pages?

File: pom.xml [hunk 4]
```diff
@@ -44,6 +65,26 @@
...
+            <version>2.3.1</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp.jstl</groupId>
+            <artifactId>javax.servlet.jsp.jstl-api</artifactId>
```
Comment (RomanDubovskyi) [inline pom.xml]: same here

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Bean
+    public LocalSessionFactoryBean getSessionFactory() {
+        LocalSessionFactoryBean factoryBean = new LocalSessionFactoryBean();
+        factoryBean.setDataSource(getDataSource());
+        Properties properties = new Properties();
+        properties.put("hibernate.show_sql", environment.getProperty("hibernate.show_sql"));
+        properties.put("hibernate.hbm2ddl.auto", environment.getProperty("hibernate.hbm2ddl.auto"));
+        factoryBean.setHibernateProperties(properties);
+        factoryBean.setAnnotatedClasses(User.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_32_2.html

File: src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public abstract class GenericDaoImpl<T> implements GenericDao<T> {
+    public static final Logger logger = Logger.getLogger(GenericDaoImpl.class);
+    SessionFactory sessionFactory;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java]: what about access modifiers?

File: src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java]: does it make sense to put this annotation on abstract class, can we create instance of it?

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImplImpl.java [hunk 8]
```diff
@@ -2,19 +2,28 @@
...
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ShoppingCartDaoImplImpl extends GenericDaoImpl<ShoppingCart>
+        implements ShoppingCartDao {
+    protected final SessionFactory sessionFactory;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImplImpl.java]: why protected?

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImplImpl.java [hunk 9]
```diff
@@ -2,19 +2,28 @@
...
+
+    public ShoppingCartDaoImplImpl(SessionFactory sessionFactory, SessionFactory sessionFactory1) {
+        super(sessionFactory);
+        this.sessionFactory = sessionFactory1;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImplImpl.java]: do we need to SessionFactories?

File: src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java [hunk 10]
```diff
@@ -9,12 +9,10 @@
 import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
-import org.springframework.stereotype.Repository;
 
-@Repository
 public abstract class GenericDaoImpl<T> implements GenericDao<T> {
     public static final Logger logger = Logger.getLogger(GenericDaoImpl.class);
-    SessionFactory sessionFactory;
+    public SessionFactory sessionFactory;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java:15]: we can use protected here and it's recommended to use as strict access modifier as possible

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**